server/database/users.js

```diff
@@ -85,6 +86,31 @@ async function getUser(username) {
   }
 }
 
+async function getUserByEmail(email) {
+  try {
+    // TODO: get this to work in a single query... so we can get rid of the emailsRef
+    const emailSnapshot = await emailsRef.child(email).once('value');
```

Comment:
You can definitely search by email on all users, but it might not be efficient.  I think that implementing it this way is the right way to go.